### ISSUE TITLE
List of real world use cases to aim for

### ISSUE THREAD

**Issue by jywarren** [publiclab/image-sequencer#707]
I'm thinking about a few goals for real world uses we've talked about in the past, and want to maintain a list of them that would be great examples of what Image Sequencer can be used for, OR what we should aim to enable it to do. 

A couple from https://github.com/publiclab/image-sequencer/issues/197#issuecomment-372354568

* particle identification and sizing for air pollution #172
* quantified color differentiation between two patches on a chemical test strip (i can add more on this later, but it's a common thing to want to do analysis!

**Comment by jywarren** [comment]
Just leaving an image re: color differentiation as an example:

![before-and-after-small](https://user-images.githubusercontent.com/24359/51771708-ad9ff480-20c8-11e9-8ccc-2bb8bf0b5c2e.png)

**Comment by HarshKhandeparkar** [comment]
@jywarren fo we want only the differentiated xolor to be visible or do we want the different in color? What do we want? Could you please elaborate?

**Comment by jywarren** [comment]
I believe for that use case (and we should break out its own issue!) we need to accept parameters for two rectangular regions to sample from, average and then display (somehow?) the difference in color -- i suppose in each channel? The rectangular regions could be standard locations -- we could say, take a picture of this card with 2 color patches, and Image Sequencer will quantify the color difference. Make sense?

I've spun the map tile filtering idea into it's own Leaflet plugin here: https://github.com/publiclab/leaflet-multispectral and submitted it to the [Leaflet Plugins listing](https://github.com/Leaflet/Leaflet/blob/master/docs/plugins.md#tileimage-display) here: https://github.com/Leaflet/Leaflet/pull/6499 -- planning to follow-up with one for tile layers as well. 

This kind of thing should be great to really show what this library is capable of and to help get the word out!

**Comment by jywarren** [comment]
The plugin was accepted and published! 

![image](https://user-images.githubusercontent.com/24359/52070994-10bddb00-2550-11e9-9313-bb6ff86a6af6.png)


I've also made a map tile filtering library, a bit more complex, and still a bit buggy -- but you can try it out here: https://publiclab.github.io/leaflet-tile-filter/

https://github.com/publiclab/leaflet-tile-filter

Once it's debugged, i'll submit it to the plugins directory too!

**Comment by HarshKhandeparkar** [comment]
@jywarren the color differenciation should take two images? One as input and one the original? I don't really know if both the regions can be in the same img. Maybe one image can be input or maybe offset? Then the color difference can be for the whole image or for the selected region? Is that how it should work? I'm not quite sure. Please elaborate a bit more. I've understood the basic concept but I don't know how the two samples will be selected. Or can you open a new issue?

**Comment by HarshKhandeparkar** [comment]
@jywarren I'll unpin this one and instead pin #694 as this one is linked inside that one.

**Comment by lohitha02** [comment]
@jywarren or @HarshKhandeparkar can someone please explain me the use case of this issue? Is it like it giving the output of a image which shows how much lead has been detected and where it has been detected(on the image)?

**Comment by HarshKhandeparkar** [comment]
This issue doesn't exactly address particle analysis, this issue addresses any general useful-in-real-life modules. That blob analysis module can be used to detect the amount of lead on the surface. That chemical spray called sprayView I guess becomes red when there is lead detected. I becomes more red when there is more lead. Lead is harmful to healt so this can be used to detect lead on samples of any general thing. May not be a particular article. May not be lead as well. It can be used to detect change in something. Specially change in physical properties like color. 

Micro particles in the air can be sized using a module like this as well. A sample can be analysed under a microscope, this module can separate particles from the background and calculate their area and approximate diameter if they were shaped like a sphere.

I hope this helps.

**Comment by HarshKhandeparkar** [comment]
Instead of using physical tools or chemical tools on every sample in real life to let's say measure concentration of lead, IS can be run on multiple of those samples at once to get the approx concentration of lead of each of those. Thanks.

**Comment by lohitha02** [comment]
Yes that was really helpful, so it's like running different modules together (like implementing the spray view thing and blob analysis module ...) and then finally giving the output of the image with the lead detected on it?  Also is the sprayView thing a separate module or something that is to be developed?

**Comment by HarshKhandeparkar** [comment]
SprayView is a real life thing not a module 😅. The picture is an example picture.

**Comment by Mridul97** [comment]
Hi @jywarren, In this color differentiation, we will accept parameters for two rectangular regions to sample from and then average the color values and display the difference in color. I have few questions. We will select the two rectangular regions from the same image or two different ones? and will this be made as a new module?

**Comment by shapironick** [comment]
just in an all-to-quick response to jeff's post on Jan 25. I would recommend a calibration patch on each colorimetric test, that helps adjust for lighting conditions and angle etc. for example over in https://github.com/publiclab/SmART-Form we have

* Calibration patch correction
	 * @param bitmap
	 * @return
	 */
	private double getRatio(Bitmap bitmap) {
		Bitmap ref, act;
		double ratio = 0;
		if(bitmap!=null) {
			act = Bitmap.createBitmap(bitmap, 50, 150, 50, 50);
			ref = Bitmap.createBitmap(bitmap, 150, 150, 50, 50);

			ratio = getLightness(act) / getLightness(ref); // Chemical badge intensity / calibrating patch intensity
		}
		return ratio;
	}

	/**
	 * Resize drawable image
	 * @param image
	 * @return
	 */
	private Drawable resize(Drawable image) {
		Bitmap b = ((BitmapDrawable)image).getBitmap();
		Bitmap bitmapResized = Bitmap.createScaledBitmap(b, 50, 50, false);
return new BitmapDrawable(getResources(), bitmapResized);

**Comment by jywarren** [comment]
I actually think this project may involve using multiple sequencers, or
otherwise using IS as a utility in the final solution, rather than the
while functionality being a module within IS, does that make sense? Kind of
like how Leaflet.Multispectral uses IS within it.

**Comment by jywarren** [comment]
So like this is something we could build using IS!

**Comment by jywarren** [comment]
Hi, I just wrote a detailed issue on the colorimetric project, with the bugs we'd need to solve, and some other ideas: https://github.com/publiclab/image-sequencer/issues/979 I hope this helps!
